Libplanet.Tests/Blockchain/BlockChainTest.cs

```diff
@@ -1774,9 +1786,25 @@ void BuildIndex(Guid id, Block<DumbAction> block)
                         b.ProtocolVersion,
                         addrs => addrs.Select(dirty.GetValueOrDefault).ToArray(),
                         (address, currency) => balances.GetValueOrDefault((address, currency)),
-                        currency => totalSupplies.TryGetValue(currency, out var totalSupply)
-                            ? totalSupply
-                            : (FungibleAssetValue?)null,
+                        currency =>
+                        {
+                            if (!currency.TotalSupplyTrackable)
+                            {
+                                var msg =
+                                    $"The total supply value of the currency {currency} is not"
+                                    + " trackable because it is a legacy untracked currency which"
+                                    + " might have been established before the introduction of"
+                                    + " total supply tracking support.";
+                                throw new TotalSupplyNotTrackableException(currency, msg);
+                            }
+
+                            if (totalSupplies.TryGetValue(currency, out var totalSupply))
+                            {
+                                return totalSupply;
+                            }
+
+                            return currency * 0;
```

Comment:
This also can be shorten:

```suggestion
                            return totalSupplies.TryGetValue(currency, out var totalSupply)
                                ? totalSupply
                                : currency * 0;
```